prog/vnet/subnet_nexus.rb

```diff
@@ -87,9 +92,20 @@ def self.random_private_ipv4(location, project, cidr_size = 26)
 
     if failure_message
       Clog.emit(failure_message)
-      selected_addr = random_private_ipv4(location, project, cidr_size)
+      return
     end
 
     selected_addr
   end
+  private_class_method :_random_private_ipv4
+
+  def self.until_random_ip(message)
+    1000.times do |i|
+      if (ip = yield)
+        return ip
+      end
+    end
+    raise message
+  end
+  private_class_method :until_random_ip
```

Comment:
For the purposes of my learning, why do we prefer meta programming here? Can we not simply follow integer passing method? I believe it would be much more easy to read the code and debug

Reply:
I found we had this same issue in `PrivateSubnet#random_private_ipv4`, so this updates the PR to use the same approach for that. This means that if you don't like the shared logic approach, we would now need to duplicate the logic in 3 places.